src/level_four/tap_cards/controller/card_action_controllers.c

```diff
@@ -66,37 +68,72 @@
 #include "wallet.h"
 #include "wallet_utilities.h"
 
-static void handle_retrieve_wallet_success(uint8_t xcor);
+/**
+ * @brief Handles post-processing required during the wallet share read
+ * operation on the X1 card.
+ *
+ * @param xcor The x-coordinate of the share contained in the X1 card
+ */
+static void read_card_share_post_process(uint8_t xcor);
 
+// TODO: This API is deprecated
 void readback_share_from_card(uint8_t xcor) {
-  tap_card_data.retries = 5;
+}
+
+bool read_card_share(uint8_t xcor, const char *heading, const char *msg) {
+  bool result = false;
+
+  card_operation_data_t card_data = {0};
+  card_data.nfc_data.retries = 5;
+  memcpy(card_data.nfc_data.family_id, get_family_id(), FAMILY_ID_SIZE);
```

Comment:
this has gone out of the `while` loop. let's keep it's position the same.

Reply:
good catch, thank you  I have reverted